SingularityService/src/main/java/com/hubspot/singularity/data/TaskManager.java

```diff
@@ -411,11 +411,19 @@ public int getNumHealthchecks(SingularityTaskId taskId) {
     return Maps.uniqueIndex(healthcheckResults, SingularityTaskIdHolder.getTaskIdFunction());
   }
 
-  @Timed
   public SingularityCreateResult saveTaskHistoryUpdate(SingularityTaskHistoryUpdate taskHistoryUpdate) {
+    return saveTaskHistoryUpdate(taskHistoryUpdate, false);
+  }
+
+  @Timed
+  public SingularityCreateResult saveTaskHistoryUpdate(SingularityTaskHistoryUpdate taskHistoryUpdate, boolean overwrite) {
     singularityEventListener.taskHistoryUpdateEvent(taskHistoryUpdate);
 
-    return create(getUpdatePath(taskHistoryUpdate.getTaskId(), taskHistoryUpdate.getTaskState()), taskHistoryUpdate, taskHistoryUpdateTranscoder);
+    if (overwrite) {
```

Comment:
As an fyi, I added this in because I had noticed that the history was no longer indicative of what killed the tasks if other manual steps were taken. For example, the cleaning would be started by a decom, but the task would actually be killed from a user request. Only task cleaning updates use the `save` instead of `create`